File: src/main/java/com/alikemal/codestat/model/XpResponse.java
package com.alikemal.codestat.model;

import java.util.List;

public class XpResponse {

    private String coded_at;
    private List<XP> xps;

    public XpResponse() {
    }

    public XpResponse(String coded_at, List<XP> xps) {
        this.coded_at = coded_at;
        this.xps = xps;
    }

    public String getCoded_at() {
        return coded_at;
    }

    public void setCoded_at(String coded_at) {
        this.coded_at = coded_at;
    }

    public List<XP> getXps() {
        return xps;
    }

    public void setXps(List<XP> xps) {
        this.xps = xps;
    }

}

File: src/main/kotlin/com/alikemal/codestat/model/XpResponse.kt
package com.alikemal.codestat.model

data class XpResponse(
    val codedAt: String,
    val xps: List<XP>
) {
    fun toJson(): String {
        val xpsJson = xps.joinToString(separator = ",", prefix = "[", postfix = "]") { it.toJson() }
        val escapedCodedAt = codedAt.replace("\\", "\\\\").replace("\"", "\\\"")
        return """{"coded_at":"$escapedCodedAt","xps":$xpsJson}"""
    }
}
